### PR TITLE
Update the build process

### PR DESCRIPTION
**Pull request by alex-wreschnig**
This would change the build process we use for deploying views and improve our ability to track changes in our view code. The build script copies from the view directory into a temporary directory and zips _that_, which means our persistent view directory can now use a generic name that we don't need to change every branch.

### CODE REVIEW

File: build_zip [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
-# just a quick thing to rebuild the zip file. Update the paths as needed.
+# just a quick thing to rebuild the zip file.
+source build_config
 
-printf "\nRebuilding zip file.\n"
-
-rm 01PITT_INST-ULS_1_8.zip
-zip -r 01PITT_INST-ULS_1_8.zip 01PITT_INST-ULS_1_8/
+# create a temporary directory
+cp -R ./view/ ./$NAME
```
Comment (ctgraham) [inline build_zip]: If `build_config` behaves poorly and doesn't set `$NAME` these commands will behave poorly.  A conditional of `[ "$NAME" != "" -a ! -e "$NAME" ]` could be beneficial here.

**Review by ctgraham — CHANGES_REQUESTED** [review]
Some sanity checking with `rm` commands is always appreciated.

File: build_zip (hunk 1 above)
Comment (alex-wreschnig) [reply build_zip]: Good call. And thanks for the sample, I wouldn't be able to put together the `-a ! -e "$NAME"` construct myself.

I've updated the build script by wrapping the filesystem-altering code in a quick conditional (borrowing your conditions) and adding a tiny amount of error reporting.

File: build_zip [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
+	printf "\nRebuilding zip file.\n"
+	rm ${NAME}.zip
+	zip -r ${NAME}.zip ./${NAME}/
+	printf "\nRebuild complete.\n"
 
-printf "\nRebuild complete.\n"
+	# remove the temporary directory
+	rm -R ./${NAME}/
+else
+	echo "Error: the temporary directory we want to create already exists."
```
Comment (ctgraham) [inline build_zip]: When spitting out error messages, best practice is to write to STDERR rather than STDOUT.  You can use `>&2 echo something` for that.

**Review by ctgraham — CHANGES_REQUESTED** [review]

File: build_zip (hunk 2 above)
Comment (alex-wreschnig) [reply build_zip]: Oh! Yes, of course. Updated...

**Review by ctgraham — APPROVED** [review]

**Pull request merged**